chore(ic-replay): [CON-1493] add more logs to `ic-replay`

kpop-dfinity opened the pull request:
Added a couple of new log lines, e.g., to know when we enter some specific if/match branches. Also added severity and timestamps to some existing logs.

Along the way renamed a function from `restore` to `restore_from_backup` which, imho, better captures the intent of the function.

Sawchord commented:
Could you give a little bit more insight into what you mean by debugging purposes?
AFAIU, in the backup tool, the logs are printed to stdout anyway, so this change only adds some context like time and severity, or am I missing something?

kpop-dfinity commented:
> Could you give a little bit more insight into what you mean by debugging purposes? AFAIU, in the backup tool, the logs are printed to stdout anyway, so this change only adds some context like time and severity, or am I missing something?

There is also a couple of extra logs I've added (mostly near the bottom of `replay.rs`) which, now I see it, in the git diff are slightly obscured by the added extra context (which is a secondary change, although timestamps can be helpful too while debugging). Sorry about that.

For example, in the test failure I'm debugging now, I suspect that we have entered [this branch](https://sourcegraph.com/github.com/dfinity/ic@d06bcccf153ba47b5016e65222fe4d5a4555dcef/-/blob/rs/replay/src/player.rs?L1022-1025), so I've added a log there. The same for [this branch](https://sourcegraph.com/github.com/dfinity/ic@6e9820533c04597a5896c8122ac0615866ca14e1/-/blob/rs/replay/src/player.rs?L1434-1439). The extra logs would confirm my hypothesis. TBH, I only really need these two logs, but I thought the other extra context is nice to have. WDYT?

Thanks for looking at the PR

Sawchord commented:
> TBH, I only really need these two logs, but I thought the other extra context is nice to have. WDYT?

Yeah I'm all in for more logging :)
I just wanted to know the background of this change
